plugin/core/types.py

```diff
@@ -440,6 +444,35 @@ def matches(self, view: sublime.View) -> bool:
         return any(f(view) for f in self.filters) if self.filters else True
 
 
+def match_file_operation_filters(
+    file_operation_options: FileOperationRegistrationOptions, path: str, view: sublime.View | None
+) -> bool:
+    def matches(file_operation_filter: FileOperationFilter) -> bool:
+        pattern = file_operation_filter.get('pattern')
+        scheme = file_operation_filter.get('scheme')
+        if scheme and view:
+            uri = view.settings().get("lsp_uri")
+            if isinstance(uri, str) and parse_uri(uri)[0] != scheme:
+                return False
+        if pattern:
+            matches = pattern.get('matches')
+            if matches:
+                if matches == FileOperationPatternKind.File and os.path.isdir(path):
+                    return False
+                if matches == FileOperationPatternKind.Folder and os.path.isfile(path):
+                    return False
+            options = pattern.get('options', {})
+            flags = GLOBSTAR | BRACE
+            if options.get('ignoreCase', False):
+                flags |= IGNORECASE
+            if not globmatch(path, pattern['glob'], flags=flags):
+                return False
+        return True
+
+    filters = [matches(file_operation_filter) for file_operation_filter in file_operation_options.get('filters')]
+    return any(filters) if filters else True
```

Comment:
This would be a bit of micro-optimization, but iirc the advantage of the `any` function is that it has short-circuit behavior, i.e. it only needs to evaluate the items until it reaches the first one which is `true`. And it accepts an iterable, so you don't need to built the whole list first, but instead can use the `for` directly inside the `any`, like

```suggestion
    return any(matches(_filter) for _filter in file_operation_options.get('filters'))
```